fire/SFEquationsMod.F90

```diff
@@ -0,0 +1,296 @@
+      !
+      !  DESCRIPTION:
+      !  Calculates reaction intensity in kJ/m2/min
+      !
+      
+      ! USES
+      use SFParamsMod, only : SF_val_fuel_energy, SF_val_miner_damp
+      
+      ! ARGUMENTS:
+      real(r8), intent(in) :: fuel_loading ! net fuel loading [kg/m2]
+      real(r8), intent(in) :: SAV          ! fuel surface area to volume ratio [/cm]
+      real(r8), intent(in) :: beta_ratio   ! ratio of packing ratio to optimum packing ratio [0-1]
+      real(r8), intent(in) :: moisture     ! fuel moisture [m3/m3]
+      real(r8), intent(in) :: MEF          ! fuel moisture of extinction [m3/m3]
+      
+      ! LOCALS:
+      real(r8) :: max_reaction_vel ! maximum reaction velocity
+      real(r8) :: opt_reaction_vel ! optimum reaction velocity
+      real(r8) :: moist_coeff      ! moisture dampening coefficient [0-1]
+      
+      ! calculate maximum reaction velocity [/min]
+      max_reaction_vel = MaximumReactionVelocity(SAV)
+      
+      ! calculate optimum reacion velocity [/min]
+      opt_reaction_vel = OptimumReactionVelocity(max_reaction_vel, SAV, beta_ratio)
+      
+      ! calculate moisture dampening coefficient [0-1]
+      moist_coeff = MoistureCoefficient(moisture, MEF)
+      
+      ReactionIntensity = opt_reaction_vel*fuel_loading*SF_val_fuel_energy*              &
+        moist_coeff*SF_val_miner_damp
+
+    end function ReactionIntensity
+  
+    !-------------------------------------------------------------------------------------
+    
+    real(r8) function HeatofPreignition(fuel_moisture)
+      !
+      !  DESCRIPTION:
+      !  Calculates heat of pre-ignition in kJ/kg
+      !
+      !  Equation A4 in Thonicke et al. 2010
+      !  Rothermel EQ12= 250 Btu/lb + 1116 Btu/lb * average_moisture
+      !  conversion of Rothermel (1972) EQ12 in BTU/lb to current kJ/kg 
+      !
+      
+      ! ARGUMENTS:
+      real(r8), intent(in) :: fuel_moisture ! fuel moisture [m3/m3]
+      
+      ! CONTANTS:
+      real(r8), parameter :: q_dry = 581.0_r8 ! heat of pre-ignition of dry fuels [kJ/kg]
+      
+      HeatofPreignition = q_dry + 2594.0_r8*fuel_moisture
+
+    end function HeatofPreignition
+
+    !-------------------------------------------------------------------------------------
+   
+    real(r8) function EffectiveHeatingNumber(SAV)
+      !
+      !  DESCRIPTION:
+      !  Calculates effective heating number [unitless]
+      !
+      !  Equation A3 in Thonicke et al. 2010
+
+      ! ARGUMENTS:
+      real(r8), intent(in) :: SAV ! fuel surface area to volume ratio [/cm]
+      
+      if (SAV < nearzero) then 
+        EffectiveHeatingNumber = 0.0_r8
+      else
+        EffectiveHeatingNumber = exp(-4.528_r8/SAV)
+      end if
+        
+    end function EffectiveHeatingNumber
+
+    !-------------------------------------------------------------------------------------
+    
+    real(r8) function PhiWind(wind_speed, beta_ratio, SAV)
```

Comment:
This function name isn't as immediately understandable as the others in this module. Could you rename to something like `WindEnhancement`?

Reply:
Can do